### ISSUE TITLE
Select all block/s with meta+a is buggy across browsers and platforms

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#7445]
Seems to me the behavior of the "Select all" shortcut `Cmd/Ctrl + A` is very buggy across different browsers and platforms, at a point that I doubt it can be considered ready to be released in its current state.

Describing what happens is a bit difficult because the actual behavior greatly varies across browsers and platforms, sometimes also with the same browser on different platforms. To my understanding, a first cause could be the way a `keydown` event works on modifier keys on different platforms. For example, in my testing on macOS when pressing _and holding_ `Cmd`, the `keydown` occurs just once. Instead, on Windows when pressing _and holding_ `Ctrl`, it occurs continuously.

Regardless, here's this is what is the test case I've used and the expected behavior:
- click on a block, for example a paragraph
- press Cmd (or Ctrl on Windows) + A: the block content is selected
- click on another paragraph
- press Cmd (or Ctrl on Windows) + A: the block content is selected
- press again Cmd (or Ctrl on Windows) + A: all the blocks get selected
- note that the last point should work either if Cmd (or Ctrl) has been hold or released and then pressed again

Seems to me this behavior works correctly only with Chrome 67 on macOS.

Firefox 60.0.2 on macOS:
- click on a block
- press Cmd + A: the block content is selected
- click on another paragraph
- press Cmd + A: the block content is selected
- press again Cmd + A: at this point looks like the block is not "selected": the UI disappears. Expected: all blocks are selected
- press Cmd + A for the third time: finally all the blocks get selected

Safari 11.1.1 (on macOS of course):
- click on a block
- press Cmd + A: the block content is selected
- click on another paragraph
- press Cmd + A: the block content is selected
- press again Cmd + A: nothing happens: the block UI stays visible and only its content is selected. Expected: all blocks should be selected

Chrome 67 on Windows 10:
- click on a block
- press Ctrl + A: the block content is selected
- click on another paragraph
- press Ctrl + A: all blocks get selected! Expected: only the block content is selected

Firefox 60.0.2 on Windows 10:
- click on a block
- press Ctrl + A: the block content is selected
- click on another paragraph
- press Ctrl + A: the block content is selected
- press again Ctrl + A: at this point looks like the block is not "selected": the UI disappears
- press Ctrl + A for the third time: finally all the blocks get selected
- until now the behavior is the same of Firefox on macOS, but...
- with all the blocks selected, click outside of them to cancel the selection
- click on a paragraph
- press press Ctrl + A: all the blocks get selected! Expected: only the block content is selected

IE 11 (similar to Chrome on Windows):
- click on a block
- press Ctrl + A: the block content is selected
- click on another paragraph
- press Ctrl + A: all blocks get selected! Expected: only the block content is selected

Sorry I can't test with Edge at the moment, because of [a recurring issue with VVV](https://github.com/Varying-Vagrant-Vagrants/VVV/issues/1523).

Considering the important impact a so buggy behavior may have on writers used to use the keyboard and users who prefer or have to use the keyboard, I'd tend to think this feature should be either fixed ore removed from the final release.

**Comment by samiff** [comment]
It looks like @afercia wasn't able to test on Edge so I thought I'd chime in. I experienced the mentioned buggy behavior:

Windows 10 Pro 64-bit (10.0, Build 17134)
Microsoft Edge 42.17134.1.0
Gutenberg Version 3.6.2 on WordPress 4.9.8

- Initial page visit, click on paragraph block.
- Press `CTRL + A` and that block's text is selected.
- Press `CTRL + A` again and all blocks are selected.
- Click in a different paragraph block.
- Press `CTRL + A` and all blocks are selected.
  - Expected the specific block's text to be selected.

I also observed other keyboard shortcuts such as `CTRL + V` for paste not working in Edge (copy did work). If I can provide any additional testing for Edge or IE 11 on Windows 10 I'd be happy to do so.

GIF:

![edge](https://user-images.githubusercontent.com/15803018/44766291-09645280-ab16-11e8-86cf-e07b2101b6d3.gif)

**Comment by designsimply** [comment]
Noting that there is an open PR about this for Safari and Firefox at #8180.

I tested pressing `meta+a` two time in a paragraph blog using WordPress 4.9.8 and Gutenberg 4.1.1 with the expectation that the first `meta+a` selects the block and the second `meta+a` selects all blocks and found the following:

* Chrome 70.0.3538.77 on macOS 10.13.6 (works as expected)
* Firefox 63.0 on macOS 10.13.6 (single block select works, requires 3 `meta+a` to select all blocks)
* Safari 12.0 on macOS 10.13.6 (select block select works, cannot select all blocks)
* Chrome 70 on Windows 10 (first `meta+a` selects all blocks, cannot select a single block)
* Firefox 63 on Windows 10 (single block select works, requires 3 `meta+a` to select all blocks)
* IE11 on Windows 10 (works as expected on a first pass, cannot select a single block alone after that)
* Edge 17 on Windows 10 (works as expected on a first pass, cannot select a single block alone after that)
